### PR TITLE
HDDS-11005. TestEndPoint#testRegisterRpcTimeout fails when run in itself

### PR DESCRIPTION
**Pull request by Tejaskriya**
## What changes were proposed in this pull request?

After running the testRegisterRpcTimeout through the flaky workflow: https://github.com/Tejaskriya/ozone/actions/runs/9476838761 , the test seems to fail each time.
Even when the single test case is run multiple times locally, failures are observed on the first run, and the remaining runs are successful. 
By seeing the logs, It seems like the steps after timeout interruption takes place could take 700-800ms when the test is run individually.
In this PR, the test is refactored to measure only the required latency of endpointTask.call(), and not the remaining steps.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-11005

## How was this patch tested?

Tested flaky workflow: https://github.com/Tejaskriya/ozone/actions/runs/9494208563

### CODE REVIEW

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java [hunk 1]
```diff
@@ -473,10 +473,10 @@ public void testRegisterNoContainerID() throws Exception {
...
   public void testRegisterRpcTimeout() throws Exception {
     final long rpcTimeout = 1000;
-    final long tolerance = 200;
-    scmServerImpl.setRpcResponseDelay(1500);
+    final long tolerance = 900;
+    scmServerImpl.setRpcResponseDelay(2500);
     long start = Time.monotonicNow();
-    registerTaskHelper(serverAddress, 1000, false).close();
+    registerTaskHelper(serverAddress, (int) rpcTimeout, false).close();
     long end = Time.monotonicNow();
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java:480]: I think the test should check elapsed time only for `endpointTask.call()`:

```java
    final int rpcTimeout = 1000;
    final int tolerance = 200;
    scmServerImpl.setRpcResponseDelay(rpcTimeout + tolerance * 2);
    OzoneConfiguration conf = SCMTestUtils.getConf(tempDir);
    try (EndpointStateMachine rpcEndPoint = createEndpoint(conf, serverAddress, rpcTimeout)) {
      rpcEndPoint.setState(EndpointStateMachine.EndPointStates.REGISTER);
      RegisterEndpointTask endpointTask = getRegisterEndpointTask(false, rpcEndPoint, conf);
      long start = Time.monotonicNow();
      endpointTask.call();
      long end = Time.monotonicNow();
      assertThat(end - start)
          .isGreaterThanOrEqualTo(rpcTimeout)
          .isLessThanOrEqualTo(rpcTimeout + tolerance);
    } finally {
      scmServerImpl.setRpcResponseDelay(0);
    }
```

We'd need some refactoring (extract methods) for that.

**Review by adoroszlai — COMMENTED** [review]
Thanks @Tejaskriya for finding this issue and working on it.

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java (hunk 1 above)
Comment (tanvipenumudy) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java:480]: > the test should check elapsed time only for endpointTask.call()

I agree, we might only need to capture the latency of the `endpointTask.call()`.

Comment (Tejaskriya) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java]: Thanks for the reviews! I have updated the PR to refactor the test. Now the latency of only endpointTask.call() is measured. Could you please review the updated patch? @adoroszlai @tanvipenumudy

**Review by adoroszlai — APPROVED** [review]

**Review by tanvipenumudy — APPROVED** [review]
Thank you @Tejaskriya for updating the patch, LGTM!

Thank you @adoroszlai for the review!

**Pull request merged**